lib/utils/lock/src/index.js

```diff
@@ -2,26 +2,33 @@
 
 // A mini wrapper around the lock module that makes it a bit easier to use.
 
-const lock = require('lock')();
+const Lock = require('lock');
 
 // Setup debug log
 const debug = require('abacus-debug')('abacus-lock');
 
-// A version of lock that uses a proper Node style callback, and passes
-// to the callback a function that can be called to unlock.
-const lockcb = (key, cb) => {
-  debug('Locking %s', key);
-  lock(key, (release) => {
-    debug('Locked %s', key);
-    cb(undefined, () => {
-      debug('Unlocking %s', key);
-      release(() => {
-        debug('Unlocked %s', key);
-      })();
+const locks = {};
+
+const locker = function(name) {
+  const n = name || 'default';
+  if(!locks[n])
+    locks[n] = new Lock();
+  const lock = locks[n];
+  const lockcb = (key, cb) => {
+    debug('Locking %s in %s locker', key, n);
+    lock(key, (release) => {
+      debug('Locked %s in %s locker', key, n);
+      cb(undefined, () => {
+        debug('Unlocking %s in %s locker', key, n);
+        release(() => {
```

Comment:
I hope I'm missing something, but we should remove from `locks` here, shouldn't we?